runipy/notebook_runner.py

```diff
@@ -194,8 +198,16 @@ def run_cell(self, cell):
                         raise NotImplementedError(
                             'unhandled mime type: %s' % mime
                         )
-
-                    setattr(out, attr, data)
+                    
+                    # In notebook version <= 3 JSON data is stored as a string
+                    # Evaluation of IPython2's JSON gives strings directly
+                    # Therefore do not encode for IPython versions prior to 3
+                    json_encode = (
```

Comment:
Yes, I'm currently experimenting with building a reporting tool based on IPython notebooks. JSON outputs are used to generate machine-readable meta-data.